handlers/spendMsgHandler.go

```diff
@@ -63,19 +59,8 @@ func NewSpendHandler(utxoStore store.UTXOStore, plasmaStore store.PlasmaStore, n
 			return sdk.ErrInternal("error updating the aggregate fee").Result()
 		}
 
-		// create new outputs
-		for i, _ := range spenderKeys {
-			utxo := store.UTXO{
-				InputKeys:        inputKeys,
-				ConfirmationHash: confirmationHash[:],
-				MerkleHash:       merkleHash,
-				Output:           spendMsg.OutputAt(uint8(i)),
-				Spent:            false,
-				Position:         positions[i],
-			}
-
-			utxoStore.StoreUTXO(ctx, utxo)
-		}
+		/* Create Outputs */
+		outputStore.StoreTx(ctx, tx)
```

Comment:
Why are you doing this twice in the same function?

Reply:
Whoops good catch